src/QMCWaveFunctions/OptimizableObject.cpp

```diff
@@ -0,0 +1,100 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2023 QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+
+#include "OptimizableObject.h"
+#include "VariableSet.h"
+#include "io/hdf/hdf_archive.h"
+
+/**@file OptimizableObject.cpp
+ *@brief OptimizableObject implementation
+ */
+
+namespace qmcplusplus
+{
+
+using opt_variables_type = optimize::VariableSet;
+
+void OptimizableObject::openHDFToSave(const std::string& filename, hdf_archive& hout)
+{
+  hout.create(filename);
+  std::vector<int> vp_file_version{2, 0, 0};
+  hout.write(vp_file_version, "version");
+}
+
+int OptimizableObject::openHDFToRead(const std::string& filename, hdf_archive& hin)
+{
+  if (!hin.open(filename, H5F_ACC_RDONLY))
+  {
+    std::ostringstream err_msg;
+    err_msg << "Unable to open VP file: " << filename;
+    throw std::runtime_error(err_msg.str());
+  }
+  std::vector<int> vp_file_version(3);
+  hin.read(vp_file_version, "version");
+  return vp_file_version[0];
+}
+
+void OptimizableObject::saveVariationalParameters(hdf_archive& hout, opt_variables_type& opt_vars)
+{
+  opt_variables_type dummy;
+  checkInVariablesExclusive(dummy);
+  hid_t name_grp = hout.push(name_);
+
+  hid_t grp = hout.push("name_value_lists");
+
+  std::vector<qmcplusplus::QMCTraits::ValueType> param_values;
+  std::vector<std::string> param_names;
+  for (auto pair_it = dummy.begin(); pair_it != dummy.end(); pair_it++)
```

Comment:
Should be have a specialization `h5data_proxy<std::map<Key,T>>`?

https://github.com/QMCPACK/qmcpack/blob/464286d39f207853a76da801301e708c4538b3a7/src/io/hdf/hdf_stl.h#L27

Reply:
VariableSet (the type of opt_vars) is not a map, though some of the API's act like it.  I'm not sure it's worth getting a map specialization for h5data_proxy working, and updating VariableSet to work with it.  VariableSet should also be replaced in the future with something that is not map-like, as part of scaling to larger numbers of variational parameters.